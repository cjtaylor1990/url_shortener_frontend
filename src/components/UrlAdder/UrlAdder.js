import React from 'react';
import './UrlAdder.css';

class UrlAdder extends React.Component {
	constructor(props) {
		super(props);

		this.state = {
			shortKey: undefined,
			fullUrl: undefined
		};

		this.handleShortKeyChange = this.handleShortKeyChange.bind(this);
		this.handleFullUrlChange = this.handleFullUrlChange.bind(this);
		this.handleAddUrlPair = this.handleAddUrlPair.bind(this);
	};

	handleShortKeyChange(event) {
		this.setState({shortKey: event.target.value});
	};

	handleFullUrlChange(event) {
		this.setState({fullUrl: event.target.value});
	};

	handleAddUrlPair(event) {
		this.props.addUrlPair(this.state.shortKey, this.state.fullUrl);
		event.preventDefault();
	};

	render() {
    	return(
			<div data-testid="urlAdder" className="urlAdder">
				<input data-testid="urlAdderFullUrl" type="text" className="fullUrl" placeholder="Original URL" onChange={this.handleFullUrlChange}/>
				<input data-testid="urlAdderShortUrl" type="text" className="shortUrl" placeholder="Short URL Key" onChange={this.handleShortKeyChange}/>
				<button data-testid="urlAdderButton" type="submit" className="addUrl" onClick={this.handleAddUrlPair}> Add URL </button>
			</div>
		);
	};

};

export default UrlAdder;
