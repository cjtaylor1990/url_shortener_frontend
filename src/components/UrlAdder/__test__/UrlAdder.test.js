import React from 'react';
import ReactDOM from 'react-dom';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';

import UrlAdder from '../UrlAdder.js';

afterEach(cleanup);

describe('The UrlAdder initially', () => {
    
    it('renders without crashing', () => {
        const div = document.createElement('div');
        
        ReactDOM.render(<UrlAdder />, div);
        
        ReactDOM.unmountComponentAtNode(div);
    });

    it('matches snapshot', () => {
        const tree = renderer.create(<UrlAdder />).toJSON();
        
        expect(tree).toMatchSnapshot();
    });

    it('to have a a pair of input boxes', () => {
        const { getByTestId } = render(<UrlAdder />);
        
        const renderedFullUrlInput = getByTestId('urlAdderFullUrl');
        const renderedShortUrlInput = getByTestId('urlAdderShortUrl');

        expect(renderedFullUrlInput).toBeInTheDocument();
        expect(renderedShortUrlInput).toBeInTheDocument();
    });

    it('the pair of input boxes are empty', () => {
        const { getByTestId } = render(<UrlAdder />);
        
        const renderedFullUrlInput = getByTestId('urlAdderFullUrl');
        const renderedShortUrlInput = getByTestId('urlAdderShortUrl');

        expect(renderedFullUrlInput).toHaveTextContent('');
        expect(renderedShortUrlInput).toHaveTextContent('');
    });
});