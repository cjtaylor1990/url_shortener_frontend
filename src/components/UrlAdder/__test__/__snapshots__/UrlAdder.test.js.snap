// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The UrlAdder initially matches snapshot 1`] = `
<div
  className="urlAdder"
  data-testid="urlAdder"
>
  <input
    className="fullUrl"
    data-testid="urlAdderFullUrl"
    onChange={[Function]}
    placeholder="Original URL"
    type="text"
  />
  <input
    className="shortUrl"
    data-testid="urlAdderShortUrl"
    onChange={[Function]}
    placeholder="Short URL Key"
    type="text"
  />
  <button
    className="addUrl"
    data-testid="urlAdderButton"
    onClick={[Function]}
    type="submit"
  >
     Add URL 
  </button>
</div>
`;
